src/controllers/AuthController.ts

```diff
@@ -0,0 +1,106 @@
+import { Router } from "../modules/Router/Router";
+import { AuthAPI, LoginData, SignupData } from "../services/API/AuthAPI";
+import store from "../modules/Store/Store";
+import { ChatsAPI } from "../services/API/ChatsAPI";
+import chatsController from "./ChatsController";
+import profileController from "./ProfileController";
+class AuthController {
+  private api: AuthAPI;
+  chatsApi: ChatsAPI;
+
+  constructor() {
+    this.api = new AuthAPI();
+    this.chatsApi = new ChatsAPI();
+  }
+
+  async signup(data: SignupData) {
+    try {
+      const registerationResponse = await this.api.signup(data);
+      if (registerationResponse.response.reason) {
+        store.set(
+          "reason",
+          `Произошла ошибка ${registerationResponse.response.reason}`
+        );
+        return;
+      }
+
+      await this.fetchUser();
+    } catch (e) {
+      console.log(e);
+    }
+  }
+
+  async login(data: LoginData) {
+    try {
+      const loginResponse = await this.api.login(data);
+      console.log(loginResponse.response);
+
+      if (loginResponse.response.reason) {
+        console.log(loginResponse.response.reason);
+
```

Comment:
Ненужное стоит подчищать
```suggestion
      if (loginResponse.response.reason) {
```